torch/_inductor/codegen/cpp.py

```diff
@@ -2798,13 +2817,22 @@ def gen_transposed_tile_load_store(self, name, var, index, is_store):
         src = f"{var} + {cexpr_index(index)}"
         dst = "__place_holder__"
         ld_src = f"{cexpr_index(stride_at_vec_range(index, self.itervars[self.tiling_idx], self.tiling_factor))}"
-        ld_dst = f"{factor}"
+        ld_dst = f"{self.num_elems}"
         if is_store:
             src, dst = dst, src
             ld_src, ld_dst = ld_dst, ld_src
 
         need_define = True
-        load_or_store = f"at::vec::transpose_mxn<{DTYPE_TO_CPP[dtype]},{factor},{factor}>({src}, {ld_src}, {dst}, {ld_dst});"
+        if self.inner_is_tiling_idx ^ is_store:
+            load_or_store = (
+                f"at::vec::transpose_mxn<{DTYPE_TO_CPP[dtype]},{self.inner_num_elems},{self.outer_num_elems}>"
+                f"({src}, {ld_src}, {dst}, {ld_dst});"
```

Comment:
Can you share the generated code when the tail size is used for generating this? BTW, does it work with dynamic shapes?

Reply:
No it dosen't work, for dynamic shapes, the tail loop still uses scalar kernel.

Reply:
I shared the generated code in the top comment.

Reply:
> No it dosen't work, for dynamic shapes, the tail loop still uses scalar kernel.

Are you going to support dynamic shapes in future PRs?

Reply:
We will consider supporting dynamic shapes in the future.